logic/peer.js

```diff
@@ -36,25 +51,35 @@ Peer.prototype.accept = function (peer) {
 		this.ip = peer.ip;
 	}
 
-	this.port = this.parseInt(peer.port, 0);
+	this.port = self.parseInt(peer.port, 0);
 
-	if (this.ip) {
-		this.string = (this.ip + ':' + this.port || 'unknown');
-	} else {
-		this.string = 'unknown';
+	if (this.ip && this.port) {
+		this.string = this.ip + ':' + this.port;
```

Comment:
also this.string should be nullified in other case- when proper port and ip are passed first and then undefined values after that `string` remains unchanged